### ISSUE TITLE
What happened to shallow clone in version 3?

### ISSUE THREAD

**Issue by pwaterz** [capistrano/capistrano#1450]
In version 2 you had the option 

```
set :git_shallow_clone, 1
```

Seems that this feature is no longer present in v3.

**Comment by leehambley** [comment]
Likely correct, don't have a facility to view the source presently, feel
free to submit a patch, I'd be happy to merge it.

**Comment by pwaterz** [comment]
Pull request https://github.com/capistrano/capistrano/pull/1454/files

**Comment by pwaterz** [comment]
It should be noted that once you move to shallow clone, you can't go back without un doing the shallow clone.

**Comment by richardwan** [comment]
another option: http://makandracards.com/makandra/1180-speed-up-capistrano-deployments-using-a-remote-cached-copy-of-repository
```
set :deploy_via, :remote_cache
set :copy_exclude, [ '.git' ]
```
this also allows you to omit additional directories that are not needed for deployment and also uses a git pull to update the repository rather than a git clone

**Comment by mattbrictson** [comment]
Closing since I believe this has been fixed by #1476.
